File: library/src/main/java/net/robinx/queue/DefaultInterceptor.java
package net.robinx.queue;

/**
 * Created by Robin on 2016/12/5 10:57.
 */

public class DefaultInterceptor<T extends Job<T>> implements Interceptor<T> {

    @Override
    public boolean interceptCondition(T job) {
        return false;
    }

    @Override
    public void onIntercept(T job) {
    }
}

File: library/src/main/java/com/github/jobqueue/DefaultInterceptor.kt
package com.github.jobqueue

/**
 * Created by Robin on 2016/12/5 10:57.
 */
class DefaultInterceptor<T : Job<T>> : Interceptor<T> {
    override fun interceptCondition(job: T): Boolean {
        return false
    }

    override fun onIntercept(job: T) {
    }
}
